### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by Salwa99**
# Vet clinic database: query and update animals table

### Heyo! :smiley: 

**In this PR I:**

- [ ] Add a column species .
- [ ] Insert the data into the table.
- [ ] Set species column to unspecified., then roll back the change .
- [ ] Inside a transaction:  Set species column to digimon then to digimon.
- [ ] Inside a transaction: Delete all animals born after Jan 1st, 2022,  create a savepoint for the transaction,  update all animals' weight to be their weight multiplied by -1, finaly, rollback to the savepoint and update all animals' weights that are negative to be their weight multiplied by -1.
- [ ] Create queries to answer the questions.
- [ ] Follow GitFlow.

### Here is the demos for those queries:
Screenshot from 2023-03-01 16-39-05

Screenshot from 2023-03-01 16-39-40

Screenshot from 2023-03-01 16-39-58

Screenshot from 2023-03-01 16-42-06

Screenshot from 2023-03-01 16-42-31

Screenshot from 2023-03-01 16-43-08

_Cheers! :star:_

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -4,5 +4,6 @@ CREATE TABLE animals (
     date_of_birth DATE,
     escape_attempts INT,
     neutered BOOLEAN,
-    weight_kg DECIMAL
+    weight_kg DECIMAL,
+    species VARCHAR(100);
```
Comment (danmainah) [inline schema.sql]: - [ ] @Salwa99 it's not good practice to edit already executed SQL command, you were supposed to use the  `ALTER TABLE` SQL command to add the new column, kindly add it correctly

File: queries.sql [hunk 2]
```diff
@@ -6,3 +6,60 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+----------------------------------------------------------------
+
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+
+ROLLBACK;
+
```
Comment (danmainah) [inline queries.sql:19]: - [ ] @Salwa99 great work on your query just like in your screenshot kindly add the `SELECT * from animals`, before and after rollback to showcase the changes
image

File: queries.sql [hunk 3]
```diff
@@ -6,3 +6,60 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+
```
Comment (danmainah) [inline queries.sql:37]: - [ ] @Salwa99 great work on your query just like in your screenshot kindly add the `SELECT * from animals`, before  rollback to showcase the changes
image

File: queries.sql [hunk 4]
```diff
@@ -6,3 +6,60 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
```
Comment (danmainah) [inline queries.sql]: - [ ] @Salwa99 great work on your query just like in your screenshot kindly add the `SELECT * from animals`, before and after commit to showcase the changes
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Salwa99 , 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: schema.sql (hunk 1 above)
Comment (Salwa99) [reply schema.sql]: Thanks @danmainah , I'll fix this minor issue

File: queries.sql (hunk 2 above)
Comment (Salwa99) [reply queries.sql:19]: @danmainah  don't you think that's too much to use every time, furthermore I respected every required step in the project.

**Comment by Salwa99** [comment]
Screenshot from 2023-03-01 17-37-52

File: queries.sql (hunk 2 above)
Comment (Kamba56) [inline queries.sql]: - [x]  As requested by the previous reviewer, kindly add the `SELECT * FROM animals;` before and after `ROLLBACK`. This is to verify changes where actually made and rolled back while running the queris.

File: queries.sql (hunk 4 above)
Comment (Kamba56) [inline queries.sql]: - [x]  As requested by the previous reviewer, kindly add the `SELECT * FROM animals;` before and after `COMMIT`. This is to verify changes were actually made and committed successfully while running the queries.

File: queries.sql [hunk 5]
```diff
@@ -6,3 +6,60 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+BEGIN;
+DELETE FROM animals;
+
+ROLLBACK;
+
```
Comment (Kamba56) [inline queries.sql:41]: - [x]  As requested by the previous reviewer, kindly add the `SELECT * FROM animals;` before and after `ROLLBACK`. This is to verify changes were actually made and rolled back while running the queries.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @Salwa99, 

Great work on making the changes requested by a previous reviewer 👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [x] Great use of Github workflow!👏👏👏
- [x] All linters passed!👏👏👏
- [ ] Good job updating the animals table using ALTER

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 5 above)
Comment (Salwa99) [reply queries.sql:41]: @Kamba56  I added it as you mentioned. Thanks!

File: queries.sql (hunk 2 above)
Comment (Abenezer-Tilahun) [reply queries.sql]: Fixed.

File: queries.sql (hunk 4 above)
Comment (Abenezer-Tilahun) [reply queries.sql]: Fixed.

File: queries.sql (hunk 5 above)
Comment (Abenezer-Tilahun) [reply queries.sql:41]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Salwa99 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git -Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**